File: src/main/java/szewek/flux/recipe/GrindingRecipe.java
package szewek.flux.recipe;

import net.minecraft.item.ItemStack;
import net.minecraft.util.ResourceLocation;
import szewek.flux.F;

public final class GrindingRecipe extends AbstractMachineRecipe {
	public GrindingRecipe(ResourceLocation idIn, MachineRecipeSerializer.Builder builder) {
		super(F.R.GRINDING, idIn, builder);
	}

	@Override
	public ItemStack getIcon() {
		return new ItemStack(F.B.GRINDING_MILL);
	}
}

File: src/main/java/szewek/flux/recipe/GrindingRecipe.kt
package szewek.flux.recipe

import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.IRecipeSerializer
import net.minecraft.util.ResourceLocation
import szewek.flux.FBlocks
import szewek.flux.FRecipes

class GrindingRecipe(idIn: ResourceLocation, groupIn: String, builder: MachineRecipeSerializer.Builder)
    : AbstractMachineRecipe(FRecipes.GRINDING, idIn, groupIn, builder) {
    override fun getIcon(): ItemStack {
        return ItemStack(FBlocks.GRINDING_MILL)
    }

    override fun getSerializer(): IRecipeSerializer<*> = FRecipes.GRINDING_SERIALIZER
}